### ISSUE TITLE
Temporary Player Methods

### ISSUE THREAD

**Issue by oloflarsson** [dmulloy2/ProtocolLib#225]
**Protocol Version:** ProtocolLib v4.0.2-SNAPSHOT-b296
**Protocol Dump:** http://pastebin.com/7dWbp6z7

Since I added ProtocolLib to my server I have had the error "The method ASDF is not supported for temporary players" appear all over the place. Here is an example: https://gyazo.com/5e45e0aa5b6dc989d392b2b9d032734f

I have tried counteracting this issue in MassiveCore a couple of times but it seems hard to cover all the situations: 
https://github.com/MassiveCraft/MassiveCore/commit/7ec9908c63caf2ffdacf491c5bda522c8b1ef77e
https://github.com/MassiveCraft/MassiveCore/commit/6caa385eb58c14b9a39b08cb8bee9ebcbc8d1160

How come these TemporaryPlayers appear? Am I doing something wrong?

Could you possibly make those temporary players implement more methods or avoid temporary players altogether?

**Comment by dmulloy2** [comment]
Yeah I agree that temporary players are not ideal. As I understand it, they're necessary because at the beginning of the login sequence all we have is the channel and no player object. They _should_ only be used until we have access to the player object.

As for dealing with them, I'd suggest avoiding caching temporary players and using their address as an identifier. I can try to add some more methods to them if you get me a list.
